### PR TITLE
fixing parsing metrics that was moved during software dragen update

### PR DESCRIPTION
**Pull request by GustavLagneborgScilifelab**
## Description
During the Dragen software update, the demultiplexing now produces certain metrics in new files with different column names. This PR fixes the parsing the these files so the output is correct

### Added

- Parsing class for Quality_metrics.csv

### Changed

-

### Fixed

- parsing from the right document
- parsing Adapter Metrics

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b fix_demux_parsing_new_dragen_version -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/apps/cgstats/parsers/quality_metrics.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    def quality_metrics_file(
+        self,
+    ) -> Dict[int, dict]:
+        """Parse the Dragen quality metrics file"""
+        LOG.info("Parsing Dragen demultiplexing quality metrics file %s", self.quality_metrics_file)
+
+        parsed_metrics = {}
+
+        with open(self.quality_metrics_path, mode="r") as metrics_file:
```
Comment (karlnyr) [inline cg/apps/cgstats/parsers/quality_metrics.py:23]:
```suggestion
        with self.quality_metrics_path.open("r") as quality_metrics_file:
```

File: cg/models/demultiplex/demux_results.py [hunk 2]
```diff
@@ -78,6 +80,13 @@ def adapter_metrics(self) -> AdapterMetrics:
         self._adapter_metrics = AdapterMetrics(self.adapter_metrics_path)
         return self._adapter_metrics
 
+    @property
+    def quality_metrics(self) -> QualityMetrics:
+        if self._quality_metrics:
+            return self._quality_metrics
+        self._quality_metrics = QualityMetrics(self.quality_metrics_path)
+        return self.__adapter_metrics
```
Comment (karlnyr) [inline cg/models/demultiplex/demux_results.py:88]:
```suggestion
        return self._adapter_metrics
```

**Review by karlnyr — COMMENTED** [review]
Think we could use a test, we could make a fixture of the `Quality_Metrics.csv` as well

**Comment by GustavLagneborgScilifelab** [comment]
dry-run test with the following command: `cg demultiplex finish flowcell -b dragen --dry-run --force 221031_A00187_0862_AHJJJNDRX2`:
```Shell
1d [gustav.lagneborg@hasta:~] [base] 2m17s 1 $ cg demultiplex finish flowcell -b dragen --dry-run --force 221031_A00187_0862_AHJJJNDRX2
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running CG demultiplex
Instantiating cgstats api
Set environment to stage
Check demuxed flowcell 221031_A00187_0862_AHJJJNDRX2
Check if demultiplexing is ready for /home/proj/stage/flowcells/novaseq/runs/221031_A00187_0862_AHJJJNDRX2
Instantiating DemuxResults with path /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2
Flowcell is already finished!
Post processing flowcell anyway
Files have already been renamed
Adding flowcell information to cgstats
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Parse log file /home/proj/stage/flowcells/novaseq/runs/221031_A00187_0862_AHJJJNDRX2/HJJJNDRX2_demultiplex.stderr
Parse log file /home/proj/stage/flowcells/novaseq/runs/221031_A00187_0862_AHJJJNDRX2/HJJJNDRX2_demultiplex.stdout
Creating new support parameters object Supportparams: /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned
Creating new datasource object Datasource: 221031_A00187_0862_AHJJJNDRX2
Creating new flowcell object <Flowcell(flowcell_id=2797, flowcellname='HJJJNDRX2', flowcell_pos='A', hiseqtype='novaseq', time=datetime.datetime(2022, 11, 1, 12, 5))>
Parsing demux conversion stats file /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned/Reports/RunInfo.xml
Creating new demux object <Demux(demux_id=3177, flowcell_id=2797, datasource_id=3352, basemask='51,8,8,51', time=datetime.datetime(2022, 11, 1, 12, 5))>
Creating new project object Project: 7049
Creating samples for flowcell 221031_A00187_0862_AHJJJNDRX2
Order samples by lane
Validate that samples are unique in lane 1
Validate that samples are unique in lane 2
Gather post Dragen demultiplexing statistics for demultiplexed samples
Parsing Dragen demultiplexing stats file /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned/Reports/Demultiplex_Stats.csv
Parsing Dragen demultiplexing quality metrics file /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned/Reports/Quality_Metrics.csv
Parsing Dragen demultiplexing adapter metrics file /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned/Reports/Adapter_Metrics.csv
Fetch report data for flowcell: HJJJNDRX2 project: 215546
Found samples: ACC10711A24,ACC10711A25,ACC10711A1,ACC10711A2,ACC10711A3,ACC10711A4,ACC10711A5,ACC10711A6,ACC10711A7,ACC10711A8,ACC10711A9,ACC10711A10,ACC10711A11,ACC10711A12,ACC10711A13,ACC10711A14,ACC10711A15,ACC10711A16,ACC10711A17,ACC10711A18,ACC10711A19,ACC10711A20,ACC10711A21,ACC10711A22,ACC10711A23,ACC10711A26,ACC10711A27,ACC10711A28,ACC10711A29,ACC10711A30,ACC10711A31,ACC10711A32,ACC10711A33,ACC10711A34,ACC10711A35,ACC10711A36,ACC10711A37,ACC10711A38,ACC10711A39,ACC10711A40,ACC10711A41,ACC10711A42,ACC10711A43,ACC10711A44,ACC10711A45,ACC10711A46 for flowcell: HJJJNDRX2, project: 215546
Write demux report to /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/stats-215546-HJJJNDRX2.txt
Copy sample sheet (/home/proj/stage/flowcells/novaseq/runs/221031_A00187_0862_AHJJJNDRX2/SampleSheet.csv) from flowcell to demuxed result dir (/home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/Unaligned/SampleSheet.csv)
Creating /home/proj/stage/demultiplexed-runs/221031_A00187_0862_AHJJJNDRX2/copycomplete.txt
```

**Comment by GustavLagneborgScilifelab** [comment]
Final test without dry-run. Outcome extpected:
Screenshot 2022-11-01 at 18 22 21

Stats file looks good:
image

**Review by karlnyr — APPROVED** [review]
good for now, needs improvement in another PR but should be merged due to time pressure

**Comment by GustavLagneborgScilifelab** [comment]
Screenshot 2022-11-01 at 18 39 36

**Comment by henrikstranneheim** [comment]
Great work @GustavLagneborgScilifelab @karlnyr

**Pull request merged**